Document/0x05b-Basic-Security_Testing.md

```diff
@@ -774,24 +773,259 @@ Note that if you have the original source code and use Android Studio, you do no
+
+##### Exploring the App Package
+
+Once you are targeting an specific app you'll want to start gathering information about it. At this point you might have the APK of the app.
+As you've seen above, installed Android apps are located at `/data/app/[package-name]` and their APKs can be retrieved from there.
+
+The APK file is an archive that contains the code and resources required to run the app it comes with. This file is identical to the original, signed app package created by the developer. It is in fact a ZIP archive with the following directory structure:
+
+```shell
+$ unzip base.apk
+$ ls -lah
+-rw-r--r--   1 sven  staff    11K Dec  5 14:45 AndroidManifest.xml
+drwxr-xr-x   5 sven  staff   170B Dec  5 16:18 META-INF
+drwxr-xr-x   6 sven  staff   204B Dec  5 16:17 assets
+-rw-r--r--   1 sven  staff   3.5M Dec  5 14:41 classes.dex
+drwxr-xr-x   3 sven  staff   102B Dec  5 16:18 lib
+drwxr-xr-x  27 sven  staff   918B Dec  5 16:17 res
+-rw-r--r--   1 sven  staff   241K Dec  5 14:45 resources.arsc
+```
+
+- AndroidManifest.xml: contains the definition of the app's package name, target and minimum [API level](https://developer.android.com/guide/topics/manifest/uses-sdk-element#ApiLevels "API Levels"), app configuration, app components, permissions, etc.
+- META-INF: contains the app's metadata
+  - MANIFEST.MF: stores hashes of the app resources
+  - CERT.RSA: the app's certificate(s)
+  - CERT.SF: list of resources and the SHA-1 digest of the corresponding lines in the MANIFEST.MF file
+- assets: directory containing app assets (files used within the Android app, such as XML files, JavaScript files, and pictures), which the [AssetManager](https://developer.android.com/reference/android/content/res/AssetManager "AssetMaanger") can retrieve
+- classes.dex: classes compiled in the DEX file format, the Dalvik virtual machine/Android Runtime can process. DEX is Java bytecode for the Dalvik Virtual Machine. It is optimized for small devices
+- lib: directory containing 3rd party libraries that are part of the APK.
+- res: directory containing resources that haven't been compiled into resources.arsc
+- resources.arsc: file containing precompiled resources, such as XML files for the layout
+
+Note that unzipping with the standard `unzip` utility the archive leaves some files unreadable. `AndroidManifest.XML` is encoded into binary XML format which isn’t readable with a text editor. Also, the app resources are still packaged into a single archive file.
+A better way of unpacking an Android app package is using [apktool](https://ibotpeaches.github.io/Apktool/). When run with default command line flags, apktool automatically decodes the Manifest file to text-based XML format and extracts the file resources (it also disassembles the .DEX files to smali code – a feature that we’ll revisit later in this book).
```

Comment:
I would move this part to the 'recommended tools - apktool' section and simply reference it here.

Reply:
that's a very nice idea. Done!